frontend/app/routes/$lang+/_protected+/personal-information+/email-address+/edit.tsx

```diff
@@ -0,0 +1,197 @@
+    { labelI18nKey: 'personal-information:email-address.edit.breadcrumbs.change-email-address' },
+  ],
+  i18nNamespaces: getTypedI18nNamespaces('personal-information', 'gcweb'),
+  pageIdentifier: pageIds.protected.personalInformation.emailAddressEdit,
+  pageTitleI18nKey: 'personal-information:email-address.edit.page-title',
+} as const satisfies RouteHandleData;
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  const instrumentationService = getInstrumentationService();
+  const raoidcService = await getRaoidcService();
+
+  await raoidcService.handleSessionValidation(request, session);
+
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformationRouteHelpers = getPersonalInformationRouteHelpers();
+  const personalInformation = await personalInformationRouteHelpers.getPersonalInformation(userInfoToken, params, request, session);
+  const emailAddress = personalInformation.emailAddress;
+
+  const csrfToken = String(session.get('csrfToken'));
+
+  const t = await getFixedT(request, handle.i18nNamespaces);
+  const meta = { title: t('gcweb:meta.title.template', { title: t('personal-information:email-address.edit.page-title') }) };
+
+  instrumentationService.countHttpStatus('email-address.edit', 200);
+  return json({ csrfToken, meta, emailAddress });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const log = getLogger('email-address/edit');
+
+  const instrumentationService = getInstrumentationService();
+  const raoidcService = await getRaoidcService();
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  await raoidcService.handleSessionValidation(request, session);
+
+  const formDataSchema = z
+    .object({
+      emailAddress: z.string(),
+      confirmEmailAddress: z.string(),
+    })
+    .superRefine((val, ctx) => {
+      if (validator.isEmpty(val.emailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.empty-email-address'), path: ['emailAddress'] });
+      } else if (!validator.isEmail(val.emailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.invalid-email-format'), path: ['emailAddress'] });
+      }
+
+      if (validator.isEmpty(val.confirmEmailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.empty-email-address'), path: ['confirmEmailAddress'] });
+      } else if (!validator.isEmail(val.confirmEmailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.invalid-email-format'), path: ['confirmEmailAddress'] });
```

Comment:
Just noticed that we don't need to use `superRefine(..)` for validating whether the input is empty or of email format. Since this form only has two email address fields, we can change the Zod schema to use Zod's built-in functions:

```typescript
  const formDataSchema = z
    .object({
      emailAddress: z
        .string()
        .trim()
        .min(1, t('email-address.edit.error-message.empty-email-address'))
        .email(t('email-address.edit.error-message.invalid-email-format')),
      confirmEmailAddress: z
        .string()
        .trim()
        .min(1, t('email-address.edit.error-message.empty-email-address'))
        .email(t('email-address.edit.error-message.invalid-email-format')),
    })
    .superRefine((val, ctx) => {
      if (val.emailAddress !== val.confirmEmailAddress) {
        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.email-match'), path: ['confirmEmailAddress'] });
      }
    });
```